Sources/Rendering/Core/CubeAxesActor/index.js

```diff
@@ -0,0 +1,831 @@
+    }
+    publicAPI.update();
+    publicAPI.modified();
+  };
+
+  // estimate from a camera model what faces to draw
+  // return true if the list of faces to draw has changed
+  publicAPI.computeFacesToDraw = () => {
+    const cmat = model.camera.getViewMatrix();
+    mat4.transpose(cmat, cmat);
+
+    let changed = false;
+    const length = vtkBoundingBox.getDiagonalLength(model.dataBounds);
+    for (let f = 0; f < 6; f++) {
+      let drawit = false;
+      const faceAxis = Math.floor(f / 2);
+      const otherAxis1 = (faceAxis + 1) % 3;
+      const otherAxis2 = (faceAxis + 2) % 3;
+      // only for non degenerate axes
+      if (
+        model.dataBounds[otherAxis1 * 2] !==
+          model.dataBounds[otherAxis1 * 2 + 1] &&
+        model.dataBounds[otherAxis2 * 2] !==
+          model.dataBounds[otherAxis2 * 2 + 1]
+      ) {
+        // for each face transform the center and off center to get a direction vector
+        ptv3[faceAxis] =
+          model.dataBounds[f] - 0.1 * length * faceNormals[f][faceAxis];
+        ptv3[otherAxis1] =
+          0.5 *
+          (model.dataBounds[otherAxis1 * 2] +
+            model.dataBounds[otherAxis1 * 2 + 1]);
+        ptv3[otherAxis2] =
+          0.5 *
+          (model.dataBounds[otherAxis2 * 2] +
+            model.dataBounds[otherAxis2 * 2 + 1]);
+        vec3.transformMat4(tmpv3, ptv3, cmat);
+        ptv3[faceAxis] = model.dataBounds[f];
+        vec3.transformMat4(tmp2v3, ptv3, cmat);
+        vec3.subtract(tmpv3, tmp2v3, tmpv3);
+        vec3.normalize(tmpv3, tmpv3);
+        // tmpv3 now holds the face normal vector
+        drawit = tmpv3[2] > 0.05;
+        // for perspctive we need the view direction to the plane
+        if (!model.camera.getParallelProjection()) {
+          drawit = vec3.dot(tmp2v3, tmpv3) > 0.05;
+        }
+      }
+      if (drawit !== model.lastFacesToDraw[f]) {
+        model.lastFacesToDraw[f] = drawit;
+        changed = true;
+      }
+    }
+    return changed;
+  };
+
+  // update the polydata that represents the boundingd edges and gridlines
+  publicAPI.updatePolyData = (facesToDraw, edgesToDraw, ticks) => {
+    // compute the number of points and lines required
+    let numPts = 0;
+    let numLines = 0;
+    numPts += 8; // always start with the 8 cube points
+
+    // count edgesToDraw
+    let numEdgesToDraw = 0;
+    for (let e = 0; e < 12; e++) {
+      if (edgesToDraw[e] > 0) {
+        numEdgesToDraw++;
+      }
+    }
+    numLines += numEdgesToDraw;
+
+    // add values for gridlines
+    if (model.gridLines) {
+      for (let f = 0; f < 6; f++) {
+        if (facesToDraw[f]) {
+          numPts +=
+            ticks[faceAxes[f][0]].length * 2 + ticks[faceAxes[f][1]].length * 2;
+          numLines +=
+            ticks[faceAxes[f][0]].length + ticks[faceAxes[f][1]].length;
+        }
+      }
+    }
+
+    // now allocate the memory
+    const points = new Float64Array(numPts * 3);
+    const lines = new Uint32Array(numLines * 3);
```

Comment:
is Uint16 enough?

Reply:
Yes, Uint16 is plenty, I'll change that up.